ansible_collections/arista/avd/roles/eos_config_deploy_cvp/templates/cvp-devices-v3.j2

```diff
@@ -11,7 +11,11 @@ CVP_DEVICES:
 {% for device in groups[container_root] | arista.avd.natural_sort %}
 {%     if device | arista.avd.is_in_filter(local_var.device_filter) %}
 {%         if hostvars[device]['is_deployed'] is arista.avd.defined(true) or hostvars[device]['is_deployed'] is not arista.avd.defined %}
-  - fqdn: {{ device }}
+{%             if cv_collection is arista.avd.defined("v3") %}
+{#                 This if block is a workaround to keep old v3 behavior of splitting hostname on "." #}
+{%                 set device = device | split('.') | first %}
+{%             endif %}
+  - hostname: {{ device }}
```

Comment:
we never used fqdns in reality, it was always hostname in both v1 and v3 (on ansible-cvp side), the key in v3 was just named incorrectly, or more correctly said whatever we input in inventory/host_vars as the device identifier we consider that as the hostname of the device regardless what it is

below is my understanding:

## behavior in v1
in v1 this is how the var looks like:

```yaml
CVP_DEVICES:
 tp-avd-leaf1:
   name: tp-avd-leaf1
   parentContainerName: TP_LEAF1
   configlets:
     - tp-avd_tp-avd-leaf1
   imageBundle: []
```

the `name` key there is populated from the inventory/host_vars in the jinja template:

https://github.com/aristanetworks/ansible-avd/blob/releases/v3.5.x/ansible_collections/arista/avd/roles/eos_config_deploy_cvp/templates/cvp-devices.j2#L13-L14

which is then sent by the v1 tasks and when it lands on ansible-cvp side where we use that as the `hostname` of the devices
few refs:
https://github.com/aristanetworks/ansible-cvp/blob/devel/ansible_collections/arista/cvp/plugins/modules/cv_device.py#L408
https://github.com/aristanetworks/ansible-cvp/blob/169074fa8993aedfc0b45b1abdd74d0db90737d4/ansible_collections/arista/cvp/plugins/modules/cv_device.py#L473
https://github.com/aristanetworks/ansible-cvp/blob/devel/ansible_collections/arista/cvp/plugins/modules/cv_device.py#L483


## behavior in v3

in v3 we have had this model:

```yaml
CVP_DEVICES:
 - fqdn: tp-avd-leaf1
   parentContainerName: TP_LEAF1
   configlets:
     - tp-avd_tp-avd-leaf1
```

however `fqdn` there is not really fqdn and in fact we set the hostname from that key (in case the user really specified an fqdn, we split it by the dots and return the first element)

https://github.com/aristanetworks/ansible-cvp/blob/169074fa8993aedfc0b45b1abdd74d0db90737d4/ansible_collections/arista/cvp/plugins/module_utils/device_tools.py#L75

with that said the v3 schema has `fqdn` or `serialNumber` keys as a must, so we cannot use `hostname` key in the j2 template today, we should set that back to `fqdn` and then update ansible-cvp later to allow `hostname` key as well
ref: https://github.com/aristanetworks/ansible-cvp/blob/devel/ansible_collections/arista/cvp/plugins/module_utils/resources/schemas/v3.py#L95-L108

I tried it as well and if we leave `hostname` the playbook fails with:

```
fatal: [cv_server]: FAILED! => changed=false
  msg: |-
    Error, your input is not valid against current schema:
     {'hostname': 'tp-avd-leaf1', 'parentContainerName': 'TP_LEAF1', 'configlets': ['tp-avd_tp-avd-leaf1']}
```